src/main/java/edu/harvard/iq/dataverse/pidproviders/doi/crossref/CrossRefDOIRegisterService.java

```diff
@@ -0,0 +1,326 @@
+                    if (author.getIdType().equals("ORCID")) {
+                        contributorsElement.append("<ORCID>").append("https://orcid.org/").append(author.getIdValue()).append("</ORCID>");
+                    }
+                    if (author.getIdType().equals("ISNI")) {
+                        contributorsElement.append("<ISNI>").append(author.getIdValue()).append("</ISNI>");
+                    }
+                    if (author.getIdType().equals("LCNA")) {
+                        contributorsElement.append("<LCNA>").append(author.getIdValue()).append("</LCNA>");
+                    }
+                }
+
+                contributorsElement.append("</person_name>");
+            }
+            contributorsElement.append("</contributors>");
+
+        } else {
+            contributorsElement.append("<contributors><person_name>")
+                    .append(AbstractPidProvider.UNAVAILABLE)
+                    .append("</person_name></contributors>");
+        }
+
+        datasetElement.append(contributorsElement);
+
+        datasetElement.append("<titles><title>")
+                .append(this.title)
+                .append("</title></titles>");
+
+        datasetElement.append("<doi_data><doi>")
+                .append(this.identifier)
+                .append("</doi><resource>")
+                .append(this.baseUrl).append("/dataset.xhtml?persistentId=doi:").append(this.identifier)
+                .append("</resource></doi_data>");
+
+        datasetElement.append("</dataset>");
+        xmlMetadata = xmlMetadata.replace("${datasets}", datasetElement.toString());
+        return xmlMetadata;
+    }
+
+    public static String getTemplate() {
+        return template;
+    }
+
+    public static void setTemplate(String template) {
+        CrossRefMetadataTemplate.template = template;
+    }
+
+    public String getIdentifier() {
+        return identifier;
+    }
+
+    public String getDepositor() {
+        return depositor;
+    }
+
+    public void setDepositor(String depositor) {
+        this.depositor = depositor;
+    }
+
+    public void setIdentifier(String identifier) {
+        this.identifier = identifier;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public void setTitle(String title) {
+        this.title = title;
+    }
+
+    public String getInstitution() {
+        return institution;
+    }
+
+    public void setInstitution(String institution) {
+        this.institution = institution;
+    }
+
+    public String getDepositorEmail() {
+        return depositorEmail;
+    }
+
+    public void setDepositorEmail(String depositorEmail) {
+        this.depositorEmail = depositorEmail;
+    }
+
+    public String getDatabaseTitle() {
+        return databaseTitle;
+    }
+
+    public void setDatabaseTitle(String databaseTitle) {
+        this.databaseTitle = databaseTitle;
+    }
+}
+
+class CrossRefFileUtil {
+
+    public static void close(InputStream in) {
```

Comment:
Looks like this is just used with readAndClose where the stream comin in is in a try-with-resources statement. Do we need any of this (the finally/close parts)?

Reply:
removed try catch finally

Reply:
So close() is no longer used anywhere?